File: src/main/java/burrow/furniture/hoard/command/PropCommand.java
package burrow.furniture.hoard.command;

import burrow.core.command.Command;
import burrow.core.command.CommandContext;
import burrow.core.command.CommandType;
import burrow.core.common.ColorUtility;
import burrow.core.common.Values;
import burrow.furniture.hoard.HoardFurniture;
import org.springframework.lang.NonNull;
import picocli.CommandLine;

@CommandLine.Command(
    name = "prop",
    description = "Retrieve the value of an entry property."
)
@CommandType(HoardFurniture.COMMAND_TYPE)
public class PropCommand extends Command {
    @CommandLine.Parameters(index = "0", description = "The ID of the entry to retrieve.")
    private Integer id;

    @CommandLine.Parameters(index = "1", description = "The key to retrieve.")
    private String key;

    public PropCommand(@NonNull final CommandContext commandContext) {
        super(commandContext);
    }

    @Override
    public Integer call() {
        final var hoardFurniture = use(HoardFurniture.class);
        final var entry = hoardFurniture.getHoard().get(id);
        final var value = entry.getOrDefault(key, null);

        if (value == null) {
            buffer.append(ColorUtility.render(Values.NULL, ColorUtility.Type.NULL));
        } else {
            buffer.append(ColorUtility.render(value, ColorUtility.Type.VALUE));
        }

        return CommandLine.ExitCode.OK;
    }
}

File: src/main/kotlin/carton/hoard/command/PropCommand.kt
package burrow.carton.hoard.command

import burrow.carton.hoard.Hoard
import burrow.carton.hoard.checkId
import burrow.kernel.terminal.*

@BurrowCommand(
    name = "prop",
    header = ["Retrieve the value of a property."]
)
class PropCommand(data: CommandData) : Command(data) {
    @Parameters(
        index = "0",
        description = [
            "The unique ID of the entry."
        ]
    )
    var id = 0

    @Parameters(
        index = "1",
        description = [
            "The key that associated with the value to retrieve."
        ]
    )
    var key = ""

    @Option(
        names = ["--left", "-l"],
        description = ["Retrieve raw property."]
    )
    var shouldRetrieveLeftValue = false

    override fun call(): Int {
        if (!checkId(id, stderr)) return ExitCode.USAGE
        if (key.isEmpty()) return ExitCode.OK

        val entry = use(Hoard::class).storage[id]
        if (entry.containsKey(key)) {
            when (shouldRetrieveLeftValue) {
                true -> stdout.println(entry.getLeft(key))
                false -> stdout.println(entry.get<Any>(key)?.toString())
            }
        }

        return ExitCode.OK
    }
}